Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs

```diff
@@ -651,3 +633,437 @@ public enum CursorVisibility {
+/// to check whether the Alt or Ctrl modifier keys were pressed when a key was pressed.
+/// </para>
+/// </summary>
+/// <remarks>
+/// <para>
+/// Lowercase alpha keys are encoded as values between 65 and 90 corresponding to the un-shifted A to Z keys on a keyboard. Enum values
+/// are provided for these (e.g. <see cref="KeyCode.A"/>, <see cref="KeyCode.B"/>, etc.). Even though the values are the same as the ASCII
+/// values for uppercase characters, these enum values represent *lowercase*, un-shifted characters.
+/// TODO: Strongly consider renaming these from .A to .Z to .A_Lowercase to .Z_Lowercase (or .a to .z).
+/// </para>
+/// <para>
+/// Numeric keys are the values between 48 and 57 corresponding to 0 to 9 (e.g. <see cref="KeyCode.D0"/>, <see cref="KeyCode.D1"/>, etc.).
+/// </para>
+/// <para>
+/// The shift modifiers (<see cref="KeyCode.ShiftMask"/>, <see cref="KeyCode.CtrlMask"/>, and <see cref="KeyCode.AltMask"/>) can be combined (with logical or)
+/// with the other key codes to represent shifted keys. For example, the <see cref="KeyCode.A"/> enum value represents the un-shifted 'a' key, while
+/// <see cref="KeyCode.ShiftMask"/> | <see cref="KeyCode.A"/> represents the 'A' key (shifted 'a' key). Likewise, <see cref="KeyCode.AltMask"/> | <see cref="KeyCode.A"/>
+/// represents the 'Alt+A' key combination.
+/// </para>
+/// <para>
+/// All other keys that produce a printable character are encoded as the Unicode value of the character. For example, the <see cref="KeyCode"/>
+/// for the '!' character is 33, which is the Unicode value for '!'. Likewise, `â` is 226, `Â` is 194, etc.
+/// </para>
+/// <para>
+/// If the <see cref="SpecialMask"/> is set, then the value is that of the special mask,
+/// otherwise, the value is the one of the lower bits (as extracted by <see cref="CharMask"/>).
+/// </para>
+/// </remarks>
+[Flags]
+public enum KeyCode : uint {
+	/// <summary>
+	/// Mask that indicates that this is a character value, values outside this range
+	/// indicate special characters like Alt-key combinations or special keys on the
+	/// keyboard like function keys, arrows keys and so on.
+	/// </summary>
+	CharMask = 0xfffff,
+
+	/// <summary>
+	/// If the <see cref="SpecialMask"/> is set, then the value is that of the special mask,
+	/// otherwise, the value is the one of the lower bits (as extracted by <see cref="CharMask"/>).
+	/// </summary>
+	SpecialMask = 0xfff00000,
+
+	/// <summary>
+	/// The key code representing null or empty
+	/// </summary>
+	Null = '\0',
+
+	/// <summary>
+	/// Backspace key.
+	/// </summary>
+	Backspace = 8,
+
+	/// <summary>
+	/// The key code for the tab key (forwards tab key).
+	/// </summary>
+	Tab = 9,
+
+	/// <summary>
+	/// The key code for the return key.
+	/// </summary>
+	Enter = '\n',
+
+	/// <summary>
+	/// The key code for the clear key.
+	/// </summary>
+	Clear = 12,
+
+	/// <summary>
+	/// The key code for the Shift key.
+	/// </summary>
+	ShiftKey = 16,
```

Comment:
If this is used as `(KeyCode)'a' | KeyCode.ShiftKey` this will result in `KeyCode.Space | KeyCode.Q` which correspond to the `q` char (`a+16`). For some reason the modifiers never was added with the real value but only as shift mask, because it will result on unexpected values. If it was only added to be uses with `ToString` method to return the correct value, that should be achieved with a different way. I also struggle with this before.
I'm not concern with the value itself but with the name that will confusing when the user want really want to the modifiers keys and use this news one.

Reply:
On Windows it is possible to have apps see the Shift key go up and down. 

I added these so i could test that and verify it was working. I don' think there's any harm in them being defined. They are not used anywhere.

Reply:
I only meant the user by mistake use them instead of the one that have mask.

Reply:
Well to say the true happened with me when I was fixing the `ConsoleKeyMapping` which by mistake I used the `ShiftKey` and the code wasn't responded as I was expected. Only on debugging I saw that I used the wrong shift key :-)